src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -7,42 +7,37 @@ internal static partial class GlobalizationMode
     {
         private static partial class Settings
         {
-            internal static readonly bool Invariant = GetGlobalizationInvariantMode();
-        }
-
-        internal static bool Invariant => Settings.Invariant;
-
-        internal static bool UseNls => false;
-
-        private static bool GetGlobalizationInvariantMode()
-        {
-            bool invariantEnabled = GetInvariantSwitchValue();
-            if (!invariantEnabled)
+            /// <summary>
+            /// Load ICU (when not in Invariant mode) in a static cctor to ensure it is loaded early in the process.
+            /// Other places, e.g. CompareInfo.GetSortKey, rely on ICU already being loaded before they are called.
+            /// </summary>
+            static Settings()
             {
-                if (TryGetAppLocalIcuSwitchValue(out string? icuSuffixAndVersion))
+                if (!Invariant)
                 {
-                    LoadAppLocalIcu(icuSuffixAndVersion);
-                }
-                else
-                {
-                    int loaded = LoadICU();
-                    if (loaded == 0 && !OperatingSystem.IsBrowser())
+                    if (TryGetAppLocalIcuSwitchValue(out string? icuSuffixAndVersion))
                     {
-                        // This can't go into resources, because a resource lookup requires globalization, which requires ICU
-                        string message = "Couldn't find a valid ICU package installed on the system. " +
-                                         "Please install libicu using your package manager and try again. " +
-                                         "Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support. " +
-                                         "Please see https://aka.ms/dotnet-missing-libicu for more information.";
-                        Environment.FailFast(message);
+                        LoadAppLocalIcu(icuSuffixAndVersion);
+                    }
+                    else
+                    {
+                        int loaded = LoadICU();
+                        if (loaded == 0)
+                        {
+                            // This can't go into resources, because a resource lookup requires globalization, which requires ICU
+                            string message = "Couldn't find a valid ICU package installed on the system. " +
+                                             "Please install libicu using your package manager and try again. " +
+                                             "Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support. " +
+                                             "Please see https://aka.ms/dotnet-missing-libicu for more information.";
```

Comment:
This message doesn't make sense for browser does it?  And the link also doesn't offer any solution.

Reply:
The message does not make sense for other workloads too. What do you think developers will do if they see it when running on Android?

Reply:
Do you have suggestions for what it should say when ICU comes with the app? Basically it means we have a bug in our code and the app was packaged incorrectly, so what should a user do in this case?

Reply:
I don't think we need to pretend we know what is wrong. A simple error message that file xyz could not be loaded would work

Reply:
or since it is not always a file  "Unable to load required ICU Globalization data. Please see https://aka.ms/dotnet-missing-libicu for more information"